moveit_core/kinematic_constraints/src/kinematic_constraint.cpp

```diff
@@ -620,9 +620,9 @@ ConstraintEvaluationResult OrientationConstraint::decide(const moveit::core::Rob
     xyz = diff.linear().eulerAngles(0, 1, 2);  // 0,1,2 corresponds to XYZ, the convention used in sampling constraints
   }
 
-  xyz(0) = std::min(fabs(xyz(0)), boost::math::constants::pi<double>() - fabs(xyz(0)));
-  xyz(1) = std::min(fabs(xyz(1)), boost::math::constants::pi<double>() - fabs(xyz(1)));
-  xyz(2) = std::min(fabs(xyz(2)), boost::math::constants::pi<double>() - fabs(xyz(2)));
+  xyz(0) = std::min(fabs(xyz(0)), boost::math::constants::two_pi<double>() - fabs(xyz(0)));
```

Comment:
Can you please put a comment above this line:  `Account for angle wrapping`

I think we could make this better by accounting for angles of k*2pi. Maybe check the remainder after division like this:

xyz(0) = std::min(fabs(xyz(0)), fmod(boost::math::constants::two_pi<double>(), fabs(xyz(0))));

Reply:
Added the comment.

I think you want the fmod applied to both sides of the min for correctness.
tmp = fmod(fabs(xyz), 2pi); xyz = min(tmp, 2pi - tmp).  Consider xyz = 1.5*pi to see why this is needed.

Leaving this to a future PR.